### ISSUE TITLE
Add a `pipeline` method to pass errors on non-scramjet streams.

### ISSUE THREAD

**Issue by MichalCz** [signicode/scramjet#21]
As raised at [openaq-fetch#561](https://github.com/openaq/openaq-fetch/issues/561) we're missing out on [a part of streams implementation in node v10 - pipeline](https://nodejs.org/api/stream.html#stream_stream_pipeline_streams_callback).

The idea here is to create a number of operations before we kick scramjet transforms in, especially pass the data through a number of Transform streams.

The common use case is `fetch |> JSONStream` or `request |> JSONStream`.

From the top of my head it should be something like:

```javascript
/**
 * Creates a pipeline of streams and returns a scramjet stream.
 *
 * @param {Array|Iterable|AsyncGeneratorFunction|GeneratorFunction|AsyncFunction|Function|String|Readable} readable
 * @param {AsyncGeneratorFunction|GeneratorFunction|AsyncFunction|Function|Transform} transforms
 */
static pipeline(readable, ...transforms) {
    const out = new this();

    (async () => {
        let current = readable;
        for (let trans of transforms) {
            current = current.pipe(trans);
            // handle errors
            // handle non-transforms, generators etc.
        }

    })().catch(e => out.raise(e));

    return out;
}
```

**Comment by MichalCz** [comment]
Initial version comitted to develop (I may regret this).

Tests missing, checks missing.

**Comment by MichalCz** [comment]
Released in v4.19.
